loki/transformations/parallel/tests/test_field_api.py

```diff
@@ -0,0 +1,129 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+import pytest
+
+from loki import Subroutine, Module, Dimension
+from loki.frontend import available_frontends, OMNI
+from loki.ir import nodes as ir, FindNodes
+from loki.logging import WARNING
+
+from loki.transformations.parallel import (
+    remove_field_api_view_updates, add_field_api_view_updates
+)
+
+
+@pytest.mark.parametrize('frontend', available_frontends(
+    skip=[(OMNI, 'OMNI needs full type definitions for derived types')]
+))
+def test_field_api_remove_view_updates(caplog, frontend):
+    """
+    A simple test for :any:`remove_field_api_view_updates`
+    """
+
+    fcode = """
+subroutine test_remove_block_loop(ngptot, nproma, nflux, dims, state, aux_fields, fluxes)
+  implicit none
+  integer(kind=4), intent(in) :: ngptot, nproma, nflux
+  type(dimension_type), intent(inout) :: dims
+  type(STATE_TYPE), intent(inout) :: state
+  type(aux_type), intent(inout) :: aux_fields
+  type(FLUX_type), intent(inout) :: fluxes(nflux)
+
+  integer :: JKGLO, IBL, ICEND, JK, JL, JF
+
+  DO jkglo=1, ngptot, nproma
+    icend = min(nproma, ngptot - JKGLO + 1)
+    ibl = (jkglo - 1) / nproma + 1
+
+    CALL DIMS%UPDATE(IBL, ICEND, JKGLO)
+    CALL STATE%update_VIEW(IBL)
+    CALL AUX_FIELDS%UPDATE_VIEW(block_index=IBL)
+    DO jf=1, nflux
+      CALL FLUXES(JF)%UPDATE_VIEW(IBL)
+    END DO
+
+    CALL MY_KERNEL(STATE%U, STATE%V, AUX_FIELDS%STUFF, FLUXES(1)%FOO, FLUXES(2)%BAR)
+  END DO
+end subroutine test_remove_block_loop
+"""
+    routine = Subroutine.from_source(fcode, frontend=frontend)
+
+    assert len(FindNodes(ir.CallStatement).visit(routine.body)) == 5
+    assert len(FindNodes(ir.Loop).visit(routine.body)) == 2
+
+    with caplog.at_level(WARNING):
+        field_group_types = ['state_type', 'aux_type', 'flux_type']
+        remove_field_api_view_updates(
+            routine, field_group_types=field_group_types, dim_object='DIMS'
+        )
+
+        assert len(caplog.records) == 2
+        assert '[Loki::ControlFlow] Removing STATE%update_VIEW call, but not in field group types!'\
+            in caplog.records[0].message
+        assert '[Loki::ControlFlow] Removing FLUXES(JF)%UPDATE_VIEW call, but not in field group types!'\
```

Comment:
I don't think there should be any warnings generated here, since the ``STATE`` and ``FLUXES`` objects are of the field group types ``STATE_TYPE`` and ``FLUX_type`` respectively, which are in the ``field_group_types``list if we don't do case-sensitive checks.